Convert CLI to dotnet tool

freakingawesome opened the pull request:
This converts the CLI into a dotnet tool (https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-tool-install) so that it can be installed via `dotnet tool install` rather than downloading binaries off of github.

Once this is compiled and pushed up to nuget.org, users can install via:

```
$ dotnet tool install --global --version 4.0.0
```

The above command would download, install, and make available a global `sq` command that can be run from anywhere, regardless of operating system.

~~BREAKING CHANGE: The assembly name of the compiled output was changed from `sq` to `Squidex.CLI` to comply with nuget standards.~~ **UPDATE** - The build output assembly name now defaults to `sq` but if you pass the build property `PackDotnetTool=1`, it will build nuget-friendly output ([see this comment](https://github.com/Squidex/squidex-samples/pull/36#issuecomment-564980785))

freakingawesome commented:
It's worth noting that in previous versions of dotnet tool, you would have to run the application via `dotnet sq ...`. However, the current iteration would let you run by just entering `sq ...` from the command line, so most of your documentation wouldn't need to change.

Also of note is that you can test installation locally by running something like the following:

```
$ dotnet tool install --global --add-source Squidex.CLI/bin/Release Squidex.CLI --version 4.0.0
```

SebastianStehle commented:
Awesome. Can we keep the assembly name so that we can also use it for environment without .NET?

freakingawesome commented:
I avoided keeping the sq.exe name because of [this warning](https://www.erikheemskerk.nl/psa-dont-change-assembly-name-published-nuget-packages/). And while that warning is specific to referenced assembly name problems, I didn't want to wander into untested territory with an executable.

Perhaps we instead supply two compilation variations, one that results in a standalone exe, and the other which generates the dotnet tool?

SebastianStehle commented:
Yes, that could work as well. Can you finish this project. I think what is left to do:

1. Find out how to fix the problem with the file name? (Perhaps we can just override it in the build process)
2. Adjust the Github Actions to publish the CLI to Nuget.

I can invite to the project if this helps.

freakingawesome commented:
Ok, I've added a few more commits to address the alternate exe name:

By default, the build will output `sq` as the assembly name, but now you can pass a build property of `PackDotnetTool=1` to cause it to use the nuget-friendly `Squidex.CLI` name. For example:

    dotnet pack -p:PackDotnetTool=1 -c Release

I've also updated the github workflow for the cli project, basing the pack/publish off of what you're doing for the client library pack/publish. It seemed pretty straight forward though I haven't fully tested it.

I haven't changed the project version but it may be worth a bump since it looks like the builds are triggered by version tags.

SebastianStehle commented:
Awesome, I will test it later today.
